src/utils/fishData.js

```diff
@@ -357,61 +796,64 @@ const fishData = {
       bodyRgb: [126, 205, 202],
       finRgb: [248, 192, 157],
       mouthRgb: [221, 148, 193],
-      knnData: [
-        normalizeColorComponent(126),
-        normalizeColorComponent(205),
-        normalizeColorComponent(202),
-        normalizeColorComponent(248),
-        normalizeColorComponent(192),
-        normalizeColorComponent(157)
-      ]
+      knnData: [126, 205, 202]
     },
     // orange
     palette2: {
       bodyRgb: [253, 192, 77],
       finRgb: [235, 120, 50],
       mouthRgb: [235, 120, 50],
-      knnData: [
-        normalizeColorComponent(253),
-        normalizeColorComponent(192),
-        normalizeColorComponent(77),
-        normalizeColorComponent(235),
-        normalizeColorComponent(120),
-        normalizeColorComponent(50)
-      ]
+      knnData: [253, 192, 77]
     },
     // red
     palette3: {
       bodyRgb: [164, 3, 31],
       finRgb: [253, 217, 136],
       mouthRgb: [253, 217, 136],
-      knnData: [
-        normalizeColorComponent(164),
-        normalizeColorComponent(3),
-        normalizeColorComponent(31),
-        normalizeColorComponent(253),
-        normalizeColorComponent(217),
-        normalizeColorComponent(136)
-      ]
+      knnData: [164, 3, 31]
     },
     // green
     palette4: {
       bodyRgb: [72, 139, 73],
       finRgb: [200, 220, 92],
       mouthRgb: [200, 220, 92],
-      knnData: [
-        normalizeColorComponent(72),
-        normalizeColorComponent(139),
-        normalizeColorComponent(73),
-        normalizeColorComponent(200),
-        normalizeColorComponent(220),
-        normalizeColorComponent(92)
-      ]
+      knnData: [72, 139, 73]
     }
   }
 };
 
-export default fishData;
+const generateKnnData = () => {
+  Object.keys(fishComponents).forEach(key => {
+    const knnDataLength = Object.values(fishComponents[key])[0].knnData.length;
+    const minArray = new Array(knnDataLength);
+    minArray.fill(Number.POSITIVE_INFINITY);
+    const maxArray = new Array(knnDataLength);
+    maxArray.fill(Number.NEGATIVE_INFINITY);
+    Object.values(fishComponents[key]).forEach(component => {
+      for (var i = 0; i < component.knnData.length; ++i) {
+        if (component.knnData[i] < minArray[i]) {
+          minArray[i] = component.knnData[i];
+        }
+        if (component.knnData[i] > maxArray[i]) {
+          maxArray[i] = component.knnData[i];
+        }
+      }
+    });
+    Object.values(fishComponents[key]).forEach(component => {
+      for (var i = 0; i < component.knnData.length; ++i) {
+        if (maxArray[i] === minArray[i]) {
+          component.knnData[i] = 0;
+        } else {
+          component.knnData[i] =
+            (component.knnData[i] - minArray[i]) / (maxArray[i] - minArray[i]);
+        }
+      }
+    });
+  });
+  return fishComponents;
+};
+
+export const fishData = generateKnnData();
```

Comment:
Should we wait until we initialise the app before running this?  Particularly if it could take more than 1/60th second, then it might be worth doing in the "loading" state.